### DISCUSSION TITLE
prisma client run in the browser error

### DISCUSSION THREAD

**Discussion by dianeDev224** [prisma/prisma#23609] [Q&A]
### Bug description

Error: PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `unknown`).

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 
PostgreSQL

### Prisma Version

```

```
lastest

**Comment by ludralph** [comment]
Hi @dianeDev224 👋 

The error `PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in unknown)` typically occurs when you try to run Prisma Client in a browser environment or if it has been bundled for the browser. Prisma Client is designed to run in a server-side environment and does not work on the client side as the browser does not execute SQL queries.

If you are using Next.js, there are a couple of solutions you can try:

1. If you are using the Next.js app router, add "use server" at the top of your file.
2. If you are using the Next.js pages router, keep the files that import Prisma or contain any Prisma query function inside the /api directory.
3. If you are trying to use Prisma in a client component, you might encounter this error. Ensure that Prisma is used in a server component

You can also take a look [here](https://github.com/prisma/prisma/issues/6219) to see how others resolved this issue.


If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
